### PR TITLE
fix: setting IDF modifier in QdrantVectorStore for sparse vectors

### LINKED ISSUE #15413: [Question]: Constructing hybrid indices with Qdrant.
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I get this error when I try to run a pipeline with qdrant as the vector store and the enable_hybrid=True. Am I missing a parameter or is this a bug?

File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/llama_index/core/instrumentation/dispatcher.py", line 260, in wrapper
    result = func(*args, **kwargs)
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/llama_index/core/ingestion/pipeline.py", line 555, in run
    self.vector_store.add(nodes_with_embeddings)
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/llama_index/vector_stores/qdrant/base.py", line 407, in add
    self._create_collection(
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/llama_index/vector_stores/qdrant/base.py", line 643, in _create_collection
    raise exc  # noqa: TRY201
  File "/home/gich2009/Work/BAYESNET/.venv/lib/python3.10/site-packages/llama_index/vector_stores/qdrant/base.py", line 616, in _create_collection
    self._client.create_collection(
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/qdrant_client/qdrant_client.py", line 2081, in create_collection
    return self._client.create_collection(
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/qdrant_client/qdrant_remote.py", line 2582, in create_collection
    sparse_vectors_config = RestToGrpc.convert_sparse_vector_config(
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/qdrant_client/conversions/conversion.py", line 3474, in convert_sparse_vector_config
    map=dict((key, cls.convert_sparse_vector_params(val)) for key, val in model.items())
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/qdrant_client/conversions/conversion.py", line 3474, in <genexpr>
    map=dict((key, cls.convert_sparse_vector_params(val)) for key, val in model.items())
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/qdrant_client/conversions/conversion.py", line 3465, in convert_sparse_vector_params
    cls.convert_modifier(model.modifier) if model.modifier is not None else None
  File "/home/gich2009/Work/.venv/lib/python3.10/site-packages/qdrant_client/conversions/conversion.py", line 3454, in convert_modifier
    raise ValueError(f"invalid Modifier model: {model}")
ValueError: invalid Modifier model: none

### PR DESCRIPTION
**Pull request by Anush008**
## Description

Fixes #15413 and also addresses https://github.com/run-llama/llama_index/issues/15413#issuecomment-2298662834

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
     HasIdCondition,
     IsEmptyCondition,
 )
+from qdrant_client.qdrant_fastembed import IDF_EMBEDDING_MODELS
```
Comment (Anush008) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py:52]: Will be an empty set if FastEmbed is not installed.

**Review by masci — APPROVED** [review]
LGTM

**Comment by masci** [comment]
@Anush008 can you bump the version in the `pyproject.toml` file so we'll release an update on merge?

File: llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py [hunk 2]
```diff
@@ -661,8 +662,8 @@ async def _acreate_collection(self, collection_name: str, vector_size: int) -> N
             index=rest.SparseIndexParams(),
             modifier=(
                 rest.Modifier.IDF
-                if self.fastembed_sparse_model and "bm42" in self.fastembed_sparse_model
-                else rest.Modifier.NONE
+                if self.fastembed_sparse_model in IDF_EMBEDDING_MODELS
+                else None
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py:666]: Just a side note, but why is rest.Modified.NONE there if it's not handled by qdrant? 👀

Comment (Anush008) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py:666]: `rest.Modifier.None` has a conversion bug when using gRPC.

`qdrant-client` 1.11.1 will fix it.

Comment (Anush008) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py:666]: https://github.com/qdrant/qdrant-client/pull/748

**Pull request merged**